q2_feature_classifier/_custom.py

```diff
@@ -0,0 +1,70 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import numpy
+from sklearn.base import BaseEstimator, ClassifierMixin, clone  # noqa
+from sklearn.utils.validation import check_X_y, check_array, check_is_fitted  # noqa
+from sklearn.naive_bayes import MultinomialNB
+from sklearn.preprocessing import LabelEncoder
+
+
+class LowMemoryMultinomialNB(MultinomialNB):
+    def __init__(self, alpha=1.0, fit_prior=True, class_prior=None,
+                 chunk_size=-1):
```

Comment:
I think it's probably ok to leave it then. I'm not terribly familiar with your generating code, but it might also be possible to check the `__annotation__` dict to see if an explicit data-type has been set. 

e.g.
```python
chunk_size: int=None):
```

That being said, I don't know if I mind the special value here, as `-1` could mean disable, and `0` *could* be automatic chunk size (even if that's not the case atm).

@jairideout what do you think?